### PR TITLE
MMP-202 Extract createMuleApp method to mule-packager module

### PR DESCRIPTION
**Pull request by WilliamPinho**

### CODE REVIEW

File: mule-maven-plugin/src/main/java/org/mule/tools/maven/mojo/PackageMojo.java [hunk 1]
```diff
@@ -95,38 +95,6 @@ protected String getFileName() {
...
-        builder.withMuleSrc(new File(targetFolder + File.separator + META_INF + File.separator + MULE_SRC));
-      }
-
-      builder.createDeployableFile();
-    } catch (IOException e) {
-      throw new MojoExecutionException("Cannot create archive");
-    }
-  }
-
   protected void initializePackageBuilder() {
```
Comment (damiansima) [inline mule-maven-plugin/src/main/java/org/mule/tools/maven/mojo/PackageMojo.java]: can we call it getPackageBuilder and no save it in the instance ?

File: mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java [hunk 2]
```diff
@@ -202,6 +205,83 @@ public void createDeployableFile() throws IOException {
...
+   * @param destinationFile file that represents the resource that is going to represent the final package.
+   * @param targetFolder folder containing the source files.
+   * @param packagingType packaging type of the app that is being built.
+   * @param onlyMuleSources when set to true, generates a package that is restricted to contain only mule sources.
+   * @param lightweightPackage when set to true, generates a package with an empty repository folder.
+   * @param attachMuleSources when set to true, adds the mule source files to final package.
+   * @throws ArchiverException
+   * @throws IOException
+   */
+  public void createMuleApp(File destinationFile, String targetFolder, PackagingType packagingType, boolean onlyMuleSources,
```
Comment (damiansima) [inline mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java]: targetFolder -> originFolder

File: mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java [hunk 3]
```diff
@@ -202,6 +205,83 @@ public void createDeployableFile() throws IOException {
     archiver.createArchive();
   }
 
+  /**
+   * Creates a mule app package based on the contents of the target folder, writing them to the destination jar file.
```
Comment (damiansima) [inline mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java]: target  folder -> origin folder

File: mule-packager/src/test/java/org/mule/tools/api/PackageBuilderTest.java [hunk 4]
```diff
@@ -39,20 +37,28 @@
...
+  private File destinationFile;
 
   @Rule
   public TemporaryFolder targetFileFolder = new TemporaryFolder();
+  @Rule
+  public TemporaryFolder destinationFileParent = new TemporaryFolder();
 
   @Before
-  public void beforeTest() {
+  public void before() throws IOException {
```
Comment (damiansima) [inline mule-packager/src/test/java/org/mule/tools/api/PackageBuilderTest.java]: it is usually called setUp

File: mule-maven-plugin/src/main/java/org/mule/tools/maven/mojo/PackageMojo.java (hunk 1 above)
Comment (WilliamPinho) [inline mule-maven-plugin/src/main/java/org/mule/tools/maven/mojo/PackageMojo.java]: Yes!

File: mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java (hunk 3 above)
Comment (WilliamPinho) [inline mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java]: Done

File: mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java (hunk 2 above)
Comment (WilliamPinho) [inline mule-packager/src/main/java/org/mule/tools/api/packager/PackageBuilder.java]: Done

**Pull request merged**